libfdt/fdt_overlay.c

```diff
@@ -816,23 +884,49 @@ int fdt_overlay_apply(void *fdt, void *fdto)
 {
 	uint32_t delta;
 	int ret;
+	uint32_t *phandlemap = NULL;
 
 	FDT_RO_PROBE(fdt);
 	FDT_RO_PROBE(fdto);
 
+	ret = fdt_find_max_phandle(fdto, &delta);
+	if (ret)
+		goto err;
+
+	phandlemap = calloc(delta, sizeof(*phandlemap));
```

Comment:
Note that even if allocations were ok in general, this could easily be a *huge* allocation.  If whatever created the overlay happened to use a single large phandle value (which is valid, though unusual) you could allocate up to 16GiB here.